src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -184,6 +194,21 @@ def collect_tool_function_in_module(m):
     return tools[0]
 
 
+def collect_tool_function_with_init_inputs_in_module(m):
+    functions = collect_tool_functions_in_module(m)
+    methods = collect_tool_methods_with_init_inputs_in_module(m)
+    num_tools = len(functions) + len(methods)
+    if num_tools == 0:
+        raise NoToolDefined("No tool found in the python script.")
+    elif num_tools > 1:
+        tool_names = ", ".join(t.__name__ for t in functions + methods)
+        raise MultipleToolsDefined(f"Expected 1 but collected {num_tools} tools: {tool_names}.")
+    if functions:
+        return functions[0], None
+    else:
+        return methods[0][0], methods[0][1]
```

Comment:
It is hard to understand methods[0][0] and methods[0][1]

Reply:
Refined.